homeassistant/components/flunearyou/sensor.py

```diff
@@ -89,10 +101,20 @@ async def async_setup_entry(hass, entry, async_add_entities):
 class FluNearYouSensor(CoordinatorEntity, SensorEntity):
     """Define a base Flu Near You sensor."""
 
-    def __init__(self, coordinator, entry, sensor_type, name, icon, unit):
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator,
+        entry: ConfigEntry,
+        sensor_type: str,
+        name: str,
+        icon: str,
+        unit: str | None,
+    ) -> None:
         """Initialize the sensor."""
         super().__init__(coordinator)
-        self._attr_extra_state_attributes = {ATTR_ATTRIBUTION: DEFAULT_ATTRIBUTION}
+        self._attr_extra_state_attributes: MutableMapping[str, Any] = {
```

Comment:
This is overriding our base typing, I think we should not do that.

Reply:
Without this re-casting, `mypy` complains:

```
homeassistant/components/recollect_waste/sensor.py:121: error: Item "None" of "Optional[MutableMapping[str, Any]]" has no attribute "update"  [union-attr]
```

Perhaps because the base entity attribute allows for `None`?

https://github.com/home-assistant/core/blob/edf42bab25ae2e3681f3d0e774b7eebdaaa87ce3/homeassistant/helpers/entity.py#L230

Reply:
With #53529 merged, if you rebase this PR, you can remove the type hint.